File: 0125-Valid-Palindrome/0125.java
import java.util.ArrayList;

class Solution {
    public boolean isPalindrome(String s) {
        ArrayList<Character> charArr = new ArrayList<>();
        s = s.toLowerCase();
        for (int i = 0; i < s.length(); i++) {
            char c = s.charAt(i);
            // conver to ascII code
            if (((int) c > 96 && (int) c < 123) || ((int) c > 47 && (int) c < 58)) {
                charArr.add(c);
            }
        }

        int left = 0;
        int right = charArr.size() - 1;

        while (left < right) {
            if (charArr.get(left) != charArr.get(right))
                return false;
            left++;
            right--;
        }

        return true;
    }
}

File: 0125-Valid-Palindrome/0125.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        str_arr = []
        s = s.lower()
        for char in s:
            # to check is Alphanumeric characters or not
            if char.isalnum():
                str_arr.append(char)
        
        # index from the left
        left = 0
        # index from the right
        right = len(str_arr) - 1
        while left < right:
            if str_arr[left] != str_arr[right]:
                return False
            left += 1
            right -= 1
        return True